### ISSUE TITLE
Update to v16 of gitlab breaks builds.

### ISSUE THREAD

**Issue by zoomix** [w4/gitlab-cargo-shim#48]
Updating gitlab to v16 from v15 started breaking builds.

https://github.com/w4/gitlab-cargo-shim/commit/915b33e3095c052e9ed5f6ce7fa45319c58c93ea

```
2023-06-28T17:10:22.692073Z  INFO ssh{peer_addr=94.254.70.114:51100 connection_id=78a96bf0-b13a-4887-8746-128344fb0c1c}: gitlab_cargo_shim: Incoming connection
2023-06-28T17:10:23.957076Z  INFO ssh{peer_addr=94.254.70.114:51100 connection_id=78a96bf0-b13a-4887-8746-128344fb0c1c}:auth_publickey{fingerprint="M97Xtvg"}: gitlab_cargo_shim: Successfully authenticated for GitLab user `srdan` by Build Token
2023-06-28T17:10:24.276620Z ERROR ssh{peer_addr=94.254.70.114:51100 connection_id=78a96bf0-b13a-4887-8746-128344fb0c1c}:data:build_packfile:fetch_token_for_user{user=User { id: 18, username: "srdan" }}: gitlab_cargo_shim::providers::gitlab: error=error decoding response body: invalid type: map, expected a string at line 1 column 11
2023-06-28T17:10:24.276647Z ERROR ssh{peer_addr=94.254.70.114:51100 connection_id=78a96bf0-b13a-4887-8746-128344fb0c1c}:data:build_packfile: gitlab_cargo_shim: error=error decoding response body: invalid type: map, expected a string at line 1 column 11
2023-06-28T17:10:24.276678Z ERROR ssh{peer_addr=94.254.70.114:51100 connection_id=78a96bf0-b13a-4887-8746-128344fb0c1c}:data: gitlab_cargo_shim: Error: error decoding response body: invalid type: map, expected a string at line 1 column 11
```

The response type map it's complaining about is the error it gets instead of the json that includes a token. 

I.e.
This requests:

```
$ curl --request POST --header "PRIVATE-TOKEN: ttcgl-" --data "name=urban_testar" --data "scopes[]=api"  "https://thetc.dev/api/v4/users/18/impersonation_tokens"
{"message":{"expires_at":["can't be blank"]}}
```

fails (400). 

While this succeeds

```
$ curl -v --request POST --header "PRIVATE-TOKEN: ttcgl-" --data "name=urban_testar" --data "scopes[]=api" --data "expires_at=2024-01-01"  "https://thetc.dev/api/v4/users/18/impersonation_tokens
{"id":13663,"...,"token":"t...","impersonation":true}%
```

So "expires_at" is mandatory, no matter what the docs say. 

As proven by this it's-late-I-gotta-go commit.
https://github.com/w4/gitlab-cargo-shim/commit/915b33e3095c052e9ed5f6ce7fa45319c58c93ea

**Comment by zoomix** [comment]
v16.1.0 to be exact

**Comment by w4** [comment]
Thanks for the report! I’ll prepare a fix shortly.

**Comment by w4** [comment]
Hi, sorry for the delay on this. The fix has been pushed in https://github.com/w4/gitlab-cargo-shim/commit/106b4ca4b36491aab217c3b5bdbc3c0685177c41 and released in [v0.1.2](https://github.com/w4/gitlab-cargo-shim/releases/tag/v0.1.2) - will be released to ghcr in [just a few minutes](https://github.com/w4/gitlab-cargo-shim/actions/runs/5448539563/jobs/9911846942)

**Comment by zoomix** [comment]
Sweet! Verified and works. Much appreciated. <3 <3 <3
